server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/LogicalPlanBuilder.java

```diff
@@ -207,22 +207,33 @@ public LogicalPlanBuilder planLast(
       Filter globalTimeFilter,
       OrderByParameter mergeOrderParameter) {
     List<PlanNode> sourceNodeList = new ArrayList<>();
-    List<PartialPath> selectedPaths =
-        sourceExpressions.stream()
-            .map(expression -> ((TimeSeriesOperand) expression).getPath())
-            .collect(Collectors.toList());
+    Map<PartialPath, List<String>> selectedPathToOutputSymbolsMap = new LinkedHashMap<>();
+    for (Expression sourceExpression : sourceExpressions) {
+      PartialPath selectedPath = ((TimeSeriesOperand) sourceExpression).getPath();
+      selectedPathToOutputSymbolsMap
+          .computeIfAbsent(selectedPath, k -> new ArrayList<>())
+          .add(sourceExpression.getOutputSymbol());
+    }
```

Comment:
That's not right, we need to make sure the order of measurement,  in this way, you break the order.